app/src/main/java/ch/epfl/polychef/HomePage.java

```diff
@@ -110,4 +124,81 @@ public boolean onNavigationItemSelected(@NonNull MenuItem selectedItem) {
                 }
         );
     }
+
+    protected void retrieveUserInfo(String email) {
+
+        getDatabase()
+                .getReference("users")
+                .orderByChild("email")
+                .equalTo(email)
+                .addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+
+                        long childrenCount = dataSnapshot.getChildrenCount();
+
+                        if(childrenCount == 0) {
+                            newUser(email);
+
+                        } else if(childrenCount == 1) {
+                            for(DataSnapshot child: dataSnapshot.getChildren()){
+                                oldUser(child);
+                            }
+
+                        } else {
+                            throw new IllegalStateException("Inconsistent result: multiple user with the same email.");
+                        }
+                    }
+
+                    @Override
+                    public void onCancelled(@NonNull DatabaseError databaseError) {
+                        //TODO: Find good exception to throw
+                        throw new IllegalArgumentException("Query cancelled");
+                    }
+                });
+    }
+
+    protected void newUser(String email) {
+        String username = getUserName();
+        user = new User(email, username);
+
+        //TODO: Integrate with the Firebase class
+        //TODO: Add OnSuccess and OnFailure listener
```

Comment:
It can be done now with the _CallHandler_ (for the onSuccess/onFailure listener part)

Reply:
Good point, we will do that in a later sprint too.